File: jikexueyuan/downloader/downloader.go
package main

import (
	"crypto/md5"
	"crypto/rand"
	"encoding/base64"
	"encoding/hex"
	"fmt"
	"io"
	"net/http"
	"os"
	// "time"
)

func getMd5String(s string) string {
	h := md5.New()
	h.Write([]byte(s))
	return hex.EncodeToString(h.Sum(nil))

}

func getGuid() string {
	b := make([]byte, 48)

	if _, err := io.ReadFull(rand.Reader, b); err != nil {
		return ""
	}
	return getMd5String(base64.URLEncoding.EncodeToString(b))
}

func main() {
	for i := 0; i < 500; i++ {
		resp, err := http.Get("http://passport.jikexueyuan.com/sso/verify?1463850163692")
		if err != nil {
			continue
		}
		fmt.Println(resp.ContentLength)
		filename := "./captchas/" + getGuid() + ".gif"
		file, err := os.Create(filename)
		if err != nil {
			continue
		}
		io.Copy(file, resp.Body)
		resp.Body.Close()
		fmt.Println(filename)
	}

}

File: jikexueyuan/downloader/downloader.py
# coding:utf-8
import requests
import uuid
from PIL import Image
import os

image_url = "http://passport.jikexueyuan.com/sso/verify"
for i in range(8000):
    try:
        resp = requests.get(image_url)
        filename = str(uuid.uuid4()) + ".gif"
        with open("./captchas/" + filename, 'wb') as f:
            for chunk in resp.iter_content(chunk_size=1024):
                if chunk:  # filter out keep-alive new chunks
                    f.write(chunk)
                    f.flush()
            f.close()
        im = Image.open("./captchas/" + filename)
        try:
            im = Image.open("./captchas/" + filename)
            im.save("./captchas/" + filename.split('.gif')[0] + ".png")
        except Exception, ex:
            print Exception, ":", ex
        os.remove("./captchas/" + filename)
        print filename
    except Exception, ex:
        print Exception, ":", ex
